CalibPPS/TimingCalibration/plugins/PPSTimingCalibrationPCLHarvester.cc

```diff
@@ -0,0 +1,160 @@
+/****************************************************************************
+ *
+ * This is a part of PPS offline software.
+ * Authors:
+ *   Edoardo Bossini
+ *   Piotr Maciej Cwiklicki
+ *   Laurent Forthomme
+ *
+ ****************************************************************************/
+
+#include "DQMServices/Core/interface/DQMEDHarvester.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include "Geometry/VeryForwardGeometryBuilder/interface/CTPPSGeometry.h"
+#include "Geometry/Records/interface/VeryForwardRealGeometryRecord.h"
+
+#include "CalibPPS/TimingCalibration/interface/TimingCalibrationStruct.h"
+#include "CondCore/DBOutputService/interface/PoolDBOutputService.h"
+
+#include "DataFormats/CTPPSDetId/interface/CTPPSDiamondDetId.h"
+#include "CondFormats/PPSObjects/interface/PPSTimingCalibration.h"
+
+#include "TProfile.h"
+#include "TF1.h"
+
+//------------------------------------------------------------------------------
+
+class PPSTimingCalibrationPCLHarvester : public DQMEDHarvester {
+public:
+  PPSTimingCalibrationPCLHarvester(const edm::ParameterSet&);
+  void beginRun(const edm::Run&, const edm::EventSetup&) override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+private:
+  void dqmEndJob(DQMStore::IBooker&, DQMStore::IGetter&) override;
+  std::vector<CTPPSDiamondDetId> detids_;
+  const std::string dqmDir_;
+  const std::string formula_;
+  const unsigned int min_entries_;
+  TF1 interp_;
+};
+
+//------------------------------------------------------------------------------
+
+PPSTimingCalibrationPCLHarvester::PPSTimingCalibrationPCLHarvester(const edm::ParameterSet& iConfig)
+    : dqmDir_(iConfig.getParameter<std::string>("dqmDir")),
+      formula_(iConfig.getParameter<std::string>("formula")),
+      min_entries_(iConfig.getParameter<unsigned int>("minEntries")),
+      interp_("interp", formula_.c_str(), 10.5, 25.) {
+  interp_.SetParLimits(1, 9., 15.);
+  interp_.SetParLimits(2, 0.2, 2.5);
+}
+
+//------------------------------------------------------------------------------
+
+void PPSTimingCalibrationPCLHarvester::beginRun(const edm::Run& iRun, const edm::EventSetup& iSetup) {
+  edm::ESHandle<CTPPSGeometry> hGeom;
+  iSetup.get<VeryForwardRealGeometryRecord>().get(hGeom);
+  for (auto it = hGeom->beginSensor(); it != hGeom->endSensor(); ++it) {
+    CTPPSDetId base_detid(it->first);
+    try {
+      CTPPSDiamondDetId detid(base_detid);
+      if (detid.station() == 1)
+        detids_.emplace_back(detid);
+    } catch (const cms::Exception&) {
+      continue;
+    }
```

Comment:
Please don't use `try`/`catch` blocks for conditions that can be checked explicitly - it should only be used to signal unexpected failures that cannot be handled locally.